drools-core/src/main/java/org/drools/core/marshalling/impl/ObjectMarshallingStrategyStoreImpl.java

```
@@ -16,15 +16,38 @@
 
 package org.drools.core.marshalling.impl;
 
+import java.util.HashSet;
+import java.util.Set;
+
+import org.drools.core.marshalling.NamedObjectMarshallingStrategy;
 import org.drools.core.util.StringUtils;
 import org.kie.api.marshalling.ObjectMarshallingStrategy;
 import org.kie.api.marshalling.ObjectMarshallingStrategyStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class ObjectMarshallingStrategyStoreImpl implements ObjectMarshallingStrategyStore {
-    private ObjectMarshallingStrategy[] strategiesList;
+	private static final Logger logger = LoggerFactory.getLogger(ObjectMarshallingStrategyStoreImpl.class);
+
+	private ObjectMarshallingStrategy[] strategiesList;
     
     public ObjectMarshallingStrategyStoreImpl(ObjectMarshallingStrategy[] strategiesList) {
         this.strategiesList = strategiesList;
+        Set<String> names = new HashSet<String>();
+        for( ObjectMarshallingStrategy strategy : strategiesList ){
+        	String name;
+        	if( strategy instanceof NamedObjectMarshallingStrategy ){
+        		name = ((NamedObjectMarshallingStrategy)strategy).getName();
+        	}else{
+        		name = strategy.getClass().getName();
+        	}
+        	if( names.contains( name ) ){
+        		logger.warn( "Multiple ObjectMarshallingStrategies with the same name:" + name + " strange behaviour could occurr");
```

if we want to add this then we should fail if there are duplicates instead of logging warning that strange behaviour can occur as that is not really helping and can lead to issues you faced and might not be easy to recover